credential_provider/pod_identity_credential_provider_test.go

```diff
@@ -1,172 +1,158 @@
-	{"Pod Identity Success via auto selection", false, false, false, preferenceAuto, ""},
-	{"Pod identity Failure via IPv4", false, false, true, preferenceIPv4, "pod identity agent returned error"},
-	{"Pod identity Failure via IPv6", false, false, true, preferenceIPv6, "pod identity agent returned error"},
-	{"Pod identity Failure via auto selection", false, false, true, preferenceAuto, "pod identity agent returned error"},
-}
-
 func TestPodIdentityCredentialProvider(t *testing.T) {
-	defer func() {
-		podIdentityAgentEndpointIPv4 = defaultIPv4Endpoint
-		podIdentityAgentEndpointIPv6 = defaultIPv6Endpoint
-	}()
-
-	for _, tstData := range podIdentityCredentialTests {
-		t.Run(tstData.testName, func(t *testing.T) {
-			provider := newPodIdentityCredentialWithMock(tstData)
-			config, err := provider.GetAWSConfig()
-
-			if len(tstData.expError) == 0 && err != nil {
-				t.Errorf("%s case: got unexpected cred provider error: %s", tstData.testName, err)
-			}
-			if len(tstData.expError) == 0 && config == nil {
-				t.Errorf("%s case: got empty config", tstData.testName)
-			}
-			if len(tstData.expError) != 0 && err == nil {
-				t.Errorf("%s case: expected error but got none", tstData.testName)
-			}
-			if len(tstData.expError) != 0 && !strings.Contains(err.Error(), tstData.expError) {
-				t.Errorf("%s case: expected error prefix '%s' but got '%s'", tstData.testName, tstData.expError, err.Error())
-			}
-		})
+	cases := []podIdentityCredentialTest{
+		{"Pod Identity Success via IPv4", false, false, false, "ipv4", ""},
+		{"Pod Identity Success via IPv6", false, false, false, "ipv6", ""},
+		{"Pod Identity Success via auto selection", false, false, false, "auto", ""},
+		{"Pod identity Failure via IPv4", false, false, true, "ipv4", "failed to refresh cached credentials, failed to load credentials, exceeded maximum number of attempts, 3, : "},
+		{"Pod identity Failure via IPv6", false, false, true, "ipv6", "failed to refresh cached credentials, failed to load credentials, exceeded maximum number of attempts, 3, : "},
+		{"Pod identity Failure via auto selection", false, false, true, "auto", "failed to refresh cached credentials, failed to load credentials, exceeded maximum number of attempts, 3, : "},
 	}
-}
-
-func TestPodIdentityToken(t *testing.T) {
-	defer func() {
-		podIdentityAgentEndpointIPv4 = defaultIPv4Endpoint
-		podIdentityAgentEndpointIPv6 = defaultIPv6Endpoint
-	}()
-
-	for _, tstData := range podIdentityTokenTests {
-
-		t.Run(tstData.testName, func(t *testing.T) {
 
-			tstAuth := newPodIdentityCredentialWithMock(tstData)
-			fetcher := tstAuth.fetcher
+	for _, tt := range cases {
+		t.Run(tt.testName, func(t *testing.T) {
+			provider, closer := newPodIdentityCredentialWithMock(t, tt.preferredEndpoint == "ipv4", tt)
+			defer func() { closer <- struct{}{} }()
 
-			tokenOut, err := fetcher.FetchToken(nil)
+			cfg, _ := provider.GetAWSConfig(context.Background())
```

Comment:
The config creation won't ever error: Getting the token is async, and happens on `cfg.Credentials.Retrieve()`